lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -225,10 +227,34 @@ sub create_from_settings {
     return $job;
 }
 
+sub rg_files {
+    my ($self, $dir, $regexp) = @_;
+    my @ret;
+    my $res = run_cmd_with_log_return_error(
+       ['rg', '--type','perl', '--type', 'py', '-l', qq('$regexp'), $dir]);
+    my $files = $res->{stderr} if $res->{stderr};
+
+    return @ret unless $files;
+
+    for my $line (split(/\n/, $files)) {
+        $line = basename $line;
+        $line =~ s/\.+$//;
+        push(@ret, $line);
+    }
+    return @ret
+}
+
+
 sub prepare_complex_query_search_args ($self, $args) {
     my @conds;
     my @joins;
 
+    if ($args->{regexp}) {
```

Comment:
A parameter "regexp" might be a bit ambiguous. It only applies to test modules. How about `module_re`?